modules/tinymce/src/plugins/lists/main/ts/core/Selection.ts

```diff
@@ -48,13 +49,27 @@ const getSelectedListItems = (editor: Editor): Array<HTMLLIElement | HTMLElement
 const getSelectedDlItems = (editor: Editor): HTMLElement[] =>
   Arr.filter(getSelectedListItems(editor), NodeType.isDlItemNode);
 
-const getClosestListRootElm = (editor: Editor, elm: Node): HTMLElement => {
+const getClosestEditingHost = (editor: Editor, elm: Node): HTMLElement => {
   const parentTableCell = editor.dom.getParents<HTMLTableCellElement>(elm, 'TD,TH');
   return parentTableCell.length > 0 ? parentTableCell[0] : editor.getBody();
 };
 
+const isListHost = (schema: Schema, node: Node) => {
+  const nodeName = node.nodeName;
+  const listNames = [ 'UL', 'OL', 'DL' ];
+
+  return !NodeType.isListNode(node) && !NodeType.isListItemNode(node) && Arr.exists(listNames, (listName) => schema.isValidChild(nodeName, listName));
+};
+
+const getClosestListHost = (editor: Editor, elm: Node): HTMLElement => {
+  const parentBlocks = editor.dom.getParents<HTMLElement>(elm, (elm) => editor.dom.isBlock(elm));
```

Comment:
```suggestion
  const parentBlocks = editor.dom.getParents<HTMLElement>(elm, editor.dom.isBlock);
```

Reply:
Hmm. Right I was forgetting that the DOM utils is a module pattern thing so there is no risk of the `this` madness happening.